docker/importer/importer.py

```diff
@@ -452,6 +453,63 @@ def thread_init():
 
     logging.info("Finished processing bucket: %s", source_repo.name)
 
+  def _process_updates_rest(self, source_repo: osv.SourceRepository):
+    """Process updates from REST API."""
+    logging.info("Begin processing REST: %s", source_repo.name)
+    import_time_now = utcnow()
+    request = requests.head(source_repo.rest_api_url, timeout=60)
+    if request.status_code != 200:
+      logging.error('Failed to fetch REST API: %s', request.status_code)
+      return
+    last_modified = datetime.datetime.strptime(request.headers['Last-Modified'],
+                                               '%a, %d %b %Y %H:%M:%S %Z')
+    # Check whether endpoint has been modified since last update
+    if last_modified < source_repo.last_update_date:
+      logging.info('No changes since last update.')
+      return
+    request = requests.get(source_repo.rest_api_url, timeout=60)
+    # Get all vulnerabilities from the REST API. (CURL approach)
+    vulns = request.json()
```

Comment:
Can you convert them into Vulnerability objects using https://github.com/google/osv.dev/blob/20deb6f580eae41473176d15789efac1381aa2b3/osv/sources.py#L123 ?

This way you can also avoid parsing the modified timestamp yourself.

Reply:
Did you give this a try? Fine to skip if this ends up being a lot slower. This can potentially just simplify/reduce some duplication of logic with timestamp parsing etc.

Reply:
Done!